python/tests/rest_api/test_rest_api.py

```diff
@@ -1,95 +1,283 @@
+import json
+import os
+
+from aac.context.definition_parser import DefinitionParser
+from tempfile import TemporaryDirectory
+from unittest.mock import patch
 from click.testing import CliRunner
 from typing import Tuple
 from unittest import TestCase
+from pydantic import BaseModel, FilePath
+
+from fastapi.encoders import jsonable_encoder
+from fastapi.testclient import TestClient
+from http import HTTPStatus
 
+from aac.context.constants import DEFINITION_FIELD_NAME, ROOT_KEY_IMPORT
+from aac.context.language_context import LanguageContext
+from aac.in_out.files.aac_file import AaCFile
+from aac.in_out.parser._parser_error import ParserError
 from aac.execute.aac_execution_result import ExecutionStatus
 from aac.execute.command_line import cli, initialize_cli
-
+from aac.in_out.parser._parse_source import parse
 from rest_api.rest_api_impl import plugin_name, rest_api, gen_openapi_spec
+from rest_api.models.command_model import (
+    CommandModel,
+    CommandRequestModel,
+    CommandResponseModel,
+    to_command_model,
+)
+from rest_api.models.definition_model import DefinitionModel, to_definition_class, to_definition_model
+from rest_api.models.file_model import FileModel, FilePathModel, FilePathRenameModel, to_file_model
+from rest_api.aac_rest_app import app, refresh_available_files_in_workspace, _get_available_files_in_workspace
+
+class TestRestApiCommands(TestCase):
+    test_client = TestClient(app)
+
+    def test_get_available_commands(self):
+        response = self.test_client.get("/commands")
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+
+    def test_execute_check_command(self):
+        command_name = "check"
+        test_model = parse(TEST_MODEL)[0]
+
+        request_arguments = CommandRequestModel(name=command_name, arguments=[TEST_MODEL, "False", "False"])
+        response = self.test_client.post("/command", data=json.dumps(jsonable_encoder(request_arguments)))
+
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertTrue(response.json().get("success"))
+        self.assertIn("success", response.text)
+        self.assertIn(command_name, response.text)
+        self.assertIn(test_model.name, response.text)
+
+    def test_execute_gen_plugin_command(self):
```

Comment:
I did miss it being a model and not a plugin, didn't think about that.  I just ran gen-plugin in the command line and it passes when used on a model, it just doesn't generate anything.  I think if I add an assert that checks if any files exist in the output dir, it will fail then.

Reply:
This test isn't really needed honestly, its purpose was just to check that commands were going through.  It can be removed.

Reply:
> I did miss it being a model and not a plugin, didn't think about that. I just ran gen-plugin in the command line and it passes when used on a model, it just doesn't generate anything. I think if I add an assert that checks if any files exist in the output dir, it will fail then.

That would be correct. Forgot that `gen_plugin` didn't have handling abut rejecting if the passed model file was not a `plugin` model, it just does not generate output. We got that type of handling in place with our `gen-puml` commands. Probably would be useful to add similar to `gen-plugin` at some point. Thoughts?

Reply:
That would probably be a good idea.

Reply:
> This test isn't really needed honestly, its purpose was just to check that commands were going through. It can be removed.

Fair, ish. I think it would be useful/good practice for us to test multiple commands. Or to have a test that confirms that all core commands are present at least.

Reply:
> > This test isn't really needed honestly, its purpose was just to check that commands were going through. It can be removed.
> 
> Fair, ish. I think it would be useful/good practice for us to test multiple commands. Or to have a test that confirms that all core commands are present at least.

So, this response comment, think you could expand the `test_get_available_commands` and confirm that all core commands are found.

Reply:
Added the checks for core aac commands.